src/server/main_service.cc

```diff
@@ -988,7 +988,8 @@ std::optional<ErrorReply> Service::VerifyCommandState(const CommandId* cid, CmdA
 
   // Check if the command is allowed to execute under this global state
   bool allowed_by_state = true;
-  switch (etl.gstate()) {
+  const auto gstate = etl.gstate();
```

Comment:
nit: can you please replace auto with GlobalState?
auto is great for generic code, avoid 120 char type names etc. But lets not use it at least for POD types - it's much harder to review the code when you do not know what type the variable is.

Reply:
done :)